mindsdb/integrations/utilities/time_series_utils.py

```diff
@@ -22,6 +23,20 @@ def transform_to_nixtla_df(df, settings_dict, exog_vars=[]):
     they must be numeric.
     """
     nixtla_df = df.copy()
+
+    # Resample every group
+    freq = settings_dict['frequency']
+    resampled_df = pd.DataFrame(columns=nixtla_df.columns)
+    if settings_dict["group_by"] and settings_dict["group_by"] != ['__group_by']:
+        for group, groupdf in nixtla_df.groupby(by=settings_dict["group_by"]):
+            groupdf.index = pd.to_datetime(groupdf.pop(settings_dict["order_by"]))
+            resampled_groupdf = pd.DataFrame(groupdf[settings_dict['target']].resample(freq).mean())
```

Comment:
This will re-sample using a moving average, which might break some of the statistical properties of the data.

Not the best-of-the-best re-sampling but should work just fine for most cases, I approve it.